### ISSUE TITLE
Preventing duplicate mutations

### ISSUE THREAD

**Issue by Shaul1321** [CodeReclaimers/neat-python#111]
Hello,

I am trying to understand the code. I would really appreciate it if someone could direct me to the section in which duplicate structural mutations are prevented (I assume that if the same mutation has occurred in two individuals in the same generation, it should get the same key).

Thanks in advance!

**Comment by drallensmith** [comment]
There is not such currently; I have put it into my draft update of the
code. (Duplicate connections will indeed get the same key already, since
their keys are determined by the two nodes they are connected to.)

-Allen

**Comment by Shaul1321** [comment]
To verify I understand - you wrote "Duplicate connections will indeed get the same key already, since their keys are determined by the two nodes they are connected to".  Doesn't this hold only for the case where the homolegous connection was formed in two individuals that happened to have the same keys for the connected nodes (i.e. the nodes were formed in the same stage, since the keys are given according to the number of node existing for that particular individuals)?

**Comment by drallensmith** [comment]
True. (An additional possibility would be if the nodes are input or output nodes.)

Versions of NEAT vary quite a bit in whether or not they have a duplicate-mutation-same-node-id mechanism, BTW.

**Comment by Shaul1321** [comment]
Thanks!
